x/msgfees/keeper/keeper.go

```diff
@@ -61,17 +61,13 @@ func (k Keeper) GetFeeCollectorName() string {
 	return k.feeCollectorName
 }
 
-func (k Keeper) GetDefaultFeeDenom() string {
-	return k.defaultFeeDenom
-}
-
-// GetFloorGasPrice  returns the current minimum gas price used in calculations for charging additional fees
-func (k Keeper) GetFloorGasPrice(ctx sdk.Context) (min uint32) {
-	min = types.DefaultFloorGasPrice
+// GetFloorGasPrice  returns the current minimum gas price in sdk.Coin used in calculations for charging additional fees
+func (k Keeper) GetFloorGasPrice(ctx sdk.Context) sdk.Coin {
+	var min uint32 = types.DefaultFloorGasPrice
```

Comment:
If this type was also a Coin it would drastically simplify some of these constructs...

Reply:
agreed..changing now

Reply:
changed